plugins/uefi/fu-plugin-uefi.c

```diff
@@ -613,40 +620,120 @@ load_custom_esp (FuPlugin *plugin, GError **error)
 
 			return FALSE;
 		}
+#ifdef HAVE_FWUP_CUSTOM_ESP
+		fwup_set_esp_mountpoint (data->esp_path);
 		g_debug ("%s set to %s", key, data->esp_path);
-		fu_plugin_add_report_metadata (plugin, key,
-					       data->esp_path);
+#endif
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_capsules (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_autofree gchar *pattern = NULL;
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* delete any files matching the glob in the ESP */
+	files = fu_common_get_files_recursive (data->esp_path, error);
+	if (files == NULL)
+		return FALSE;
+	pattern = g_build_filename (data->esp_path, "EFI/*/fw/fwupdate-*.cap", NULL);
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch (pattern, fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
 	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_efivars (FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GPtrArray) files = NULL;
 
+	/* delete any efivars matching the glob */
+	files = fu_common_get_files_recursive ("/sys/firmware/efi/efivars", error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch ("fwupdate-*", fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
```

Comment:
Have you tested this on a real machine at all?  I think this might actually fail since the efivars files are marked immutable by default on current kernels.

Reply:
An alternative is to use `efivar` and `efi_del_variable`.

Reply:
I think efi_del_variable is a much better idea indeed. Willfix.

Reply:
Fixed, thanks for the idea.